src/serializer/visitors.js

```diff
@@ -116,6 +135,23 @@ export let ClosureRefReplacer = {
     }
   },
 
+  // TODO: handle FunctionDeclaration
+  FunctionExpression(path: BabelTraversePath, state: ClosureRefReplacerState) {
+    if (t.isProgram((path: any).parentPath.parentPath.node)) {
+      // Skip root function itself.
+      // This assumes the root function is wrapped with: t.file(t.program([t.expressionStatement(rootFunction).
+      return;
```

Comment:
Okay, its about the "root function". But I don't understand why we need this special handling.

Reply:
Our goal is traversing the residual function and replace any duplicated "nested" function. We do not want to replace residual function itself, right? Otherwise, we may replace the root residual function with bind(null) to itself which is wrong.